src/coreclr/jit/emitarm64.cpp

```diff
@@ -2294,6 +2306,1470 @@ emitter::code_t emitter::emitInsCode(instruction ins, insFormat fmt)
+            for (index = 0; index < 2; index++)
+            {
+                if (fmt == formatEncode2BK[index])
+                {
+                    encoding_found = true;
+                    break;
+                }
+            }
+            break;
+        case IF_SVE_2BL:
+            for (index = 0; index < 2; index++)
+            {
+                if (fmt == formatEncode2BL[index])
+                {
+                    encoding_found = true;
+                    break;
+                }
+            }
+            break;
+        case IF_SVE_2BM:
+            for (index = 0; index < 2; index++)
+            {
+                if (fmt == formatEncode2BM[index])
+                {
+                    encoding_found = true;
+                    break;
+                }
+            }
+            break;
+        case IF_SVE_2BN:
+            for (index = 0; index < 2; index++)
+            {
+                if (fmt == formatEncode2BN[index])
+                {
+                    encoding_found = true;
+                    break;
+                }
+            }
+            break;
+        case IF_SVE_2BO:
+            for (index = 0; index < 2; index++)
+            {
+                if (fmt == formatEncode2BO[index])
+                {
+                    encoding_found = true;
+                    break;
+                }
+            }
+            break;
+        case IF_SVE_2BP:
+            for (index = 0; index < 2; index++)
+            {
+                if (fmt == formatEncode2BP[index])
+                {
+                    encoding_found = true;
+                    break;
+                }
+            }
+            break;
+        case IF_SVE_2BQ:
+            for (index = 0; index < 2; index++)
+            {
+                if (fmt == formatEncode2BQ[index])
+                {
+                    encoding_found = true;
+                    break;
+                }
+            }
+            break;
+        case IF_SVE_2BR:
+            for (index = 0; index < 2; index++)
+            {
+                if (fmt == formatEncode2BR[index])
+                {
+                    encoding_found = true;
+                    break;
+                }
+            }
+            break;
+        case IF_SVE_2BS:
+            for (index = 0; index < 2; index++)
+            {
+                if (fmt == formatEncode2BS[index])
+                {
+                    encoding_found = true;
+                    break;
+                }
+            }
+            break;
+        default:
+            if (fmt == insFmt)
+            {
+                encoding_found = true;
+                index          = 0;
+            }
+            else
+            {
+                encoding_found = false;
+            }
+            break;
+    }
+
+    assert(encoding_found);
+
+    switch (index)
+    {
+        case 0:
+            assert(ins < ArrLen(insCodes1));
+            code = insCodes1[ins];
```

Comment:
To avoid this switch, there could be a NxM array insCodes which each insCodes[0] is a pointer to insCodes1,  insCodes[1] is a pointer to  insCodes2 etc

Then instead of the switch you can access via insCodes[index][ins].

Reply:
Yes, I think there are multiple ways to optimize this (as we spoke offline). I will probably have them as a separate change.